packages/op-coreutils/src/op_coreutils/bigquery/write.py

```diff
@@ -152,3 +161,55 @@ def _write_df_to_bq(df: pl.DataFrame, destination: str, job_config=bigquery.Load
         log.info(
             f"{operation_prefix}{job_config.write_disposition}: Wrote {human_rows(len(df))} {human_size(filesize)} to BQ {destination}"
         )
+
+
+def upsert_partition(
+    df: pl.DataFrame, dt: str, dataset: str, table_name: str, unique_keys=["dt"]
+):
+    """Upsert data into a specific partition in a BigQuery table.
```

Comment:
I think there's something to clarify here. Do we expect the upsert method to be used with a partitioned table always? From the name of the function it sounds like it.  

Also what would be the expectations on the incoming df, should it have only 1 `dt` value or will the function deal with it possibly containing several `dt` values?

Reply:
> Do we expect the upsert method to be used with a partitioned table always?

Good point, I increased the scope of the function and turned it into `upsert_table`. It should now handle both cases